python/lsst/ts/salobj/sal_info.py

```diff
@@ -235,31 +245,41 @@ def parse_idl(self):
         * event_names: a tuple of event names, without the ``"logevent_"``
           prefix
         * telemetry_names: a tuple of telemetry topic names
+        * sal_topic_names: a tuple of SAL topic names, e.g.
+          "logevent_summaryState", in alphabetical order
         * revnames: a dict of topic name: name_revision
         """
-        pattern = re.compile(r" *struct +(?P<name_rev>(?P<name>.+)_(?:[a-zA-Z0-9]+)) +{ *")
+        struct_pattern = re.compile(r"\s*struct\s+(?P<name_rev>(?P<name>.+)_(?:[a-zA-Z0-9]+)) +{")
+        index_pattern = re.compile(rf"\s*long\s+{self.name}ID;")
```

Comment:
We found one parser for OMG IDL but it is not maintained and it breaks horribly on our IDL. It's tricky to do well because there are so many vendor extensions.

Another option would be to pick a parsing library and use that. But this is simple and it works. I don't need much information from the IDL file (though more than I used to, due to the added checks).